engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodNode.java

```diff
@@ -235,6 +237,65 @@ Stateful doConvertText(
     }
   }
 
+  @Specialization(
+      guards = {
+        "!methods.hasFunctionalDispatch(_this)",
+        "!methods.hasSpecialDispatch(_this)",
+        "getPolyglotCallType(_this, symbol.getName(), interop) == CONVERT_TO_DATE"
+      })
+  Stateful doConvertDate(
+      VirtualFrame frame,
+      Object state,
+      UnresolvedSymbol symbol,
+      Object _this,
+      Object[] arguments,
+      @CachedLibrary(limit = "10") MethodDispatchLibrary methods,
+      @CachedLibrary(limit = "1") MethodDispatchLibrary dateDispatch,
+      @CachedLibrary(limit = "10") InteropLibrary interop,
+      @Cached("wrapDateInAtom(interop)") WrapDateInAtom wrapDate
+  ) {
+    try {
+      Atom date = wrapDate.newDate(_this);
+      Function function = dateDispatch.getFunctionalDispatch(date, symbol);
+      arguments[0] = date;
+      return invokeFunctionNode.execute(function, frame, state, arguments);
+    } catch (MethodDispatchLibrary.NoSuchMethodException e) {
+      throw new PanicException(
+          Context.get(this).getBuiltins().error().makeNoSuchMethodError(_this, symbol), this);
+    }
+  }
+
+  static final class WrapDateInAtom extends Node {
+    private final AtomConstructor dateConstructor;
```

Comment:
This is wrong! We cannot cache `Atom` neither `AtomConstructor` in a `Node`, right @kustosz? I'll rewrite the code and put the cache into `Context`...

Reply:
Rewritten in e13400118e212e91db8e07bbdda2ad90a1984d15 to store the `AtomConstructor` in `Context`.